ground/gcs/src/plugins/uploader/uploadergadgetwidget.cpp

```diff
@@ -1469,48 +1470,51 @@ void UploaderGadgetWidget::onBootloaderRemoved()
     }
 }
 
+void UploaderGadgetWidget::startRescue()
+{
+    setStatusInfo(tr("Trying to detect bootloader"), uploader::STATUSICON_INFO);
+    m_widget->progressBar->setValue(100);
+
+    connect(&rescueTimer, &QTimer::timeout, this,
+            &UploaderGadgetWidget::onRescueTimer, Qt::UniqueConnection);
+    rescueTimer.start(200);
+
+    connect(usbFilterBL, &USBSignalFilter::deviceDiscovered,
+            this, &UploaderGadgetWidget::onDeviceDiscovered, Qt::UniqueConnection);
+    connect(usbFilterUP, &USBSignalFilter::deviceDiscovered,
+            this, &UploaderGadgetWidget::onDeviceDiscovered, Qt::UniqueConnection);
+}
+
 /**
  * @brief slot called by a timer created by itself or by the onRescueButton
  * @param start true if the rescue timer is to be started
  */
-void UploaderGadgetWidget::onRescueTimer(bool start)
+void UploaderGadgetWidget::onRescueTimer()
```

Comment:
Refactored this into separate functions (as it should have been before IMO). Was required to avoid signal/slot parameter incompatibility.